internal/compiler/typeloader.rs

```diff
@@ -36,12 +36,14 @@ pub struct ImportedName {
 }
 
 impl ImportedName {
-    pub fn extract_imported_names(
-        import: &syntax_nodes::ImportSpecifier,
-    ) -> Option<impl Iterator<Item = ImportedName>> {
-        import
-            .ImportIdentifierList()
-            .map(|import_identifiers| import_identifiers.ImportIdentifier().map(Self::from_node))
+    pub fn extract_imported_names(imports: &[syntax_nodes::ImportSpecifier]) -> Vec<ImportedName> {
+        let mut imported_names = Vec::new();
+        for import in imports {
+            if let Some(import_identifiers) = import.ImportIdentifierList() {
+                imported_names.extend(import_identifiers.ImportIdentifier().map(Self::from_node));
+            }
+        }
+        imported_names
```

Comment:
I tried for a while to express this using flat_map, but I failed. Since you pointed it out, I tried again, and I found a combination now - I was missing `Option::into_iter` :)

Reply:
maybe fliter_map followed by flat_map

Reply:
I couldn't get `filter_map` to work, but I can move `into_iter()`. I find the former better than the latter though:

Current:
```rust
        imports
            .iter()
            .flat_map(|import| {
                import.ImportIdentifierList().into_iter().flat_map(|import_identifiers| {
                    import_identifiers.ImportIdentifier().map(Self::from_node)
                })
            })
            .collect::<Vec<_>>()
```

With moved `into_iter()`:

```rust
        imports
            .iter()
            .flat_map(|import| {
                import
                    .ImportIdentifierList()
                    .map(|import_identifiers| {
                        import_identifiers.ImportIdentifier().map(Self::from_node)
                    })
                    .into_iter()
                    .flatten()
            })
            .collect::<Vec<_>>()
```